curvefs/src/client/s3/disk_cache_manager.cpp

```diff
@@ -337,27 +337,27 @@ void DiskCacheManager::TrimCache() {
         SetDiskFsUsedRatio();
         if (IsDiskCacheFull()) {
             VLOG(3) << "disk cache full, begin trim.";
-            std::string cacheReadFullDir;
-            std::string cacheWriteFullDir;
+            std::string cacheReadFullDir, cacheWriteFullDir,
+              cacheReadFile, cacheWriteFile;
             cacheReadFullDir = GetCacheReadFullDir();
             cacheWriteFullDir = GetCacheWriteFullDir();
-
+            std::string cacheKey, cacheKeyBfo, cacheKeyDel;
+            if (!cachedObjName_->GetBack(&cacheKey)) {
+                VLOG(3) << "remove disk file error"
+                        << ", cachedObjName is empty.";
+                continue;
+            }
             while (!IsDiskCacheSafe()) {
```

Comment:
use 'do while' will be better, just remove the object from  GetBack before judge IsDiskCacheSafe and GetBefore.

Reply:
GetBefore should before remove object